dietpi/dietpi-drive_manager

```diff
@@ -998,17 +1000,35 @@
 
 				fi
 
-			elif [ "$G_WHIP_RETURNED_VALUE" = "Move User data" ]; then
+			elif [ "$G_WHIP_RETURNED_VALUE" = 'Reserved blocks' ]; then
+
+				G_WHIP_INPUTBOX 'Ext4 formatted drives allow the reservation of drive space for the root user to assure system functionality, if filled by other users or processes, and to avoid fragmentation of large files.\n\nHowever, on modern drives, the default of 5% reserved blocks is often by orders of magnitude larger than necessary. You may want to reduce the percentage to an absulte reserved space of about 500 MiB, which should be enough, to enable root user starting and maintaining the system. Additionally, on non rootfs drives reserved blocks are not necessary at all.\n\nPlease enter the desired percentage of reserved blocks, e.g. "0.05" for 0.05% or "10" for 10%.\nNote: Only values between "0" and "50" are allowed.'
+				if (( ! $? )); then
+
+					if grep -qE '^[0-9]+\.?[0-9]*$' <<< "$G_WHIP_RETURNED_VALUE" &&
```

Comment:
We should be aiming to use global here:
```
G_CHECK_VALIDINT
```

We should tweak ```G_CHECK_VALIDINT``` if needed?

Reply:
Maybe G_CKECK_VALIDNUM and allow a variable/option to allow floats as well, which are is needed here?

Would be also nice if we find a more elegant way to compare float values, than sed it into integers :smile:

Reply:
@Fourdee 
Jep possible as well, maybe separate checks are more secure/cleaner. Just to have reference, though the solutions are similar to mine: https://stackoverflow.com/a/13790836
- But will we need this more often, that a global function already make sense? 😉

Reply:
@Fourdee 
grep -qE '^[0-9]*\.?[0-9]*$' works fine as float test, as ".5" == "0.5" and "5." == "5.0" is also accepted.
But I am not sure if this is true for all commands that generally accept floats. If we want to create G_CHECK_VALIDFLOAT, then perhaps it's better to just allow values with filled zero from our end. 

Also:
root@VM-Stretch:~# bc -l <<< '05 == 5'
1
root@VM-Stretch:~# tune2fs -m 05 /dev/sda1
tune2fs 1.43.4 (31-Jan-2017)
Setting reserved blocks percentage to 5% (104844 blocks)

Leading zeros are allowed by check as above and just ignored by bc and tune2fs, but this might be not true as well for other commands. Quite difficult to test, if we don't want to allow leading zeros, and want to have it, if followed by decimal dot...